Training24/Program.c

```diff
@@ -1,15 +1,45 @@
-// ------------------------------------------------------------------------------------
+// --------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// -------------------------------------
 //  Dhaarani Mohanasundaram
-// -------------------------------------
-// Standard Program file in Main
-// -------------------------------------
+// --------------------------------------------------------------------------------
 // Program.c
-// Program on main branch.
-// ------------------------------------------------------------------------------------
+// Program on branch A4.
+// Checks for phrase palindrome and Reverse the number then checks for palindrome.
+// --------------------------------------------------------------------------------
+#include <ctype.h>
+#include <stdbool.h>
+#include <limits.h>
 
-const char* GetMessage() {
-return "Hello, World!";
+bool CheckPhrasePalindrome (const char* phrase) {
+   if (phrase == NULL) return 0;   // Not a palindrome
+   char onlyAlphaNum[256] = "";
+   int j = 0, start = 0, end = j - 1;
+   // Filter out non-alphanumeric characters and convert to lowercase
+   for (int i = 0; phrase[i] != '\0'; i++) if (isalnum (phrase[i])) onlyAlphaNum[j++] = tolower (phrase[i]);
+   onlyAlphaNum[j] = '\0';
+   while (start < end) {
+      if (onlyAlphaNum[start] != onlyAlphaNum[end]) return false;
+      start++;
+      end--;
+   }
+   return true;
 }
+
+bool ReverseAndCheckOverflow (int number, int* reversedNumber) {
```

Comment:
Your method is for reversal and that is what its name should indicate. If you are checking for overflow, you can have a separate bit for it or return false (since operation would not have been separately carried out.